### DISCUSSION TITLE
Lamellae spacing in 2D

### DISCUSSION THREAD

**Discussion by jeanmichelscherer** [neperfepx/neper#621] [Q&A] [answered]
Hi,

I would like to create a 2D 2-scale tessellation in which parent (voronoi) grains are divided into equally-spaced lamellae. I would like the parent grains to share the same crystallographic direction along the plane normal (`fiber(2,-1,-1,0,0,1)`).
I tried the following:
```
neper -T -n 50::from_morpho -morpho 'voronoi::lamellar(w=0.02,v=crysdir(1,1,1))'\
         -ori 'fiber(2,-1,-1,0,0,1)::fiber(2,-1,-1,0,0,1)'\
         -oriformat 'plain' -oridescriptor 'euler-bunge:passive' -dim 2\
         -format tess,geo,ori  -id 0  -domain 'square(1.0,1.0)'\
         -o meshes/n50-id0_rcl0-25_lamellar_child
```
The lamellae normal direction is `crysdir(1,1,1)` and it is chosen to be in the plane of normal `(2,-1,-1)`. 
Do you know why this command results in different spacings of the lamellae in each parent grain? 
I would understand this behaviour if the normal was not in the fiber plane.

Thanks for your help,
Jean-Michel

![img](https://user-images.githubusercontent.com/53854997/206807245-5db3d772-f98a-4499-bf83-d1875fe9cd79.png)

**Comment by mkasemer** [comment, marked as answer]
EDIT: I see now you have specified the fiber out of plane. Let me inspect further.

EDIT 2: Apologies for my initial response. Indeed, I interpet it the same as you: you force the crystal plane to be in the plane of the 2D sample, and thus the lamellae should all share the same thickness since their normal is in the spatial 2D plane. Have you attempted to manually confirm this from the set of orientations? That is, for the orientations, have you attempted to transform the plane normal vectors to see if they are all in plane?

EDIT 3: I went ahead and tested your above command and got the following (using `neper -V n50-id0_rcl0-25_lamellar_child.tess -print test`). Are you using an outdated version?
![test](https://user-images.githubusercontent.com/2902614/206824249-d1bcaf63-58f5-47a6-b9a9-6cc5ea943b26.png)

**Reply by jeanmichelscherer** [reply]
Yes, I did something of this kind. From the set of orientations of the parent grains, I plotted the direction of the `(1,0,0)`-spatial axis in the crystal frame. All vectors lie on the same disc which has `(2,-1,-1)`-crystal as normal.
![Figure_1](https://user-images.githubusercontent.com/53854997/206824494-3779e1ec-12b4-4a81-9e3b-18138c907c41.png)

**Reply by jeanmichelscherer** [reply]
I am on version 4.2.1-27. I'll do an update.

**Reply by jeanmichelscherer** [reply]
I get the same result as yours after the update to the latest version 4.5.1-4. Thank you for the help!
